src/utils/cloudflare/api.js

```diff
@@ -0,0 +1,114 @@
+const assert = require('assert');
+const retry = require('promise-retry');
+const { helpers: { generateClass } } = require('common-errors');
+
+const { CloudflareClient } = require('./client');
+
+const RULE_LIST = 'accounts/:accountId/rules/lists';
+const RULE_ITEM_LIST = 'accounts/:accountId/rules/lists/:listId/items';
+const RULE_BATCH_PROGESS = 'accounts/:accountId/rules/lists/bulk_operations/:operationId';
+
+const BulkOperationError = generateClass('BulkOperationError', {
+  args: ['message', 'response'],
+});
+
+const defaultRetryConfig = {
+  retries: 10,
+  factor: 0.5,
+};
+
+class CloudflareAPI {
+  constructor(cfClient, config = {}, withProps = {}) {
+    assert(cfClient instanceof CloudflareClient, 'invalid client');
```

Comment:
this wont allow stubbing during tests, so I'd add a condition that this is not a test environment

Reply:
Removed `instanceof` check, the constructor now checks the existence of parameter.